moto/emr/models.py

```diff
@@ -200,34 +201,43 @@ def __init__(
             "master_instance_type" in instance_attrs
             and instance_attrs["master_instance_type"]
         ):
-            self.emr_backend.add_instance_groups(
+            instance_groups = [
+                {
+                    "instance_count": 1,
+                    "instance_role": "MASTER",
+                    "instance_type": instance_attrs["master_instance_type"],
+                    "market": "ON_DEMAND",
+                    "name": "master",
+                }
+            ]
+            instance_group_result = self.emr_backend.add_instance_groups(
                 self.id,
-                [
-                    {
-                        "instance_count": 1,
-                        "instance_role": "MASTER",
-                        "instance_type": instance_attrs["master_instance_type"],
-                        "market": "ON_DEMAND",
-                        "name": "master",
-                    }
-                ],
+                instance_groups,
+            )
+            self.emr_backend.add_instances(
```

Comment:
Personally not a huge fan of mutating `self.emr_backend` inside the initializer here, but we're following the same pattern as `add_instance_groups(..)` which is already in here, so should be fine. 👍 

(We may need to think about how this influences persistence/reloading /cc @giograno , as these calls in the initializer may have undesired side effects for our persistence logic..)